src/app/confirmation/[confirmationKey]/page.js

```diff
@@ -0,0 +1,42 @@
+"use client";
+
+import { useEffect, useState } from 'react';
+import { useRouter } from "next/navigation";
+import { useMutation } from "react-query";
+import { Error } from "@/app/activities/[activityId]/edit/components/formikFields";
+import { Container, CircularProgress } from '@mui/material';
+import { verifyEmail } from "@/app/api.mjs";
+
+export default function EmailConfirmation({ params }) {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+  const mutation = useMutation(verifyEmail);
+  const { confirmationKey } = params;
+  const decodedKey = decodeURIComponent(confirmationKey);
+
+  const confirmEmail = () => {
```

Comment:
again, naming is not consistent with api. I'd name it handleVerifyEmail at least. As max, I would just write whole operation in useEffect.